pkg/controller/migplan/migplan_controller.go

```diff
@@ -434,6 +451,50 @@ func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	return nil
 }
 
+func (r *ReconcileMigPlan) AddUpdateLabelFailedMigration(plan *migapi.MigPlan) error {
+
+	failedMigrations := 0
+	needsUpdate := false
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	//Check if there are any associated migrations that are failed
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.PlanNotReady) {
```

Comment:
hrm, is this the right condition for Failed Migration? how about https://github.com/konveyor/mig-controller/blob/4b11788223c4aa5394e0d4881ca0ee8d3e55bc64/pkg/controller/migmigration/validation.go#L19?

Reply:
Updated the conditions